x/txfees/keeper/grpc_query.go

```diff
@@ -0,0 +1,48 @@
+package keeper
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/x/txfees/types"
+)
+
+var _ types.QueryServer = Keeper{}
+
+func (k Keeper) FeeTokens(ctx context.Context, _ *types.QueryFeeTokensRequest) (*types.QueryFeeTokensResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	feeTokens := k.GetFeeTokens(sdkCtx)
+
+	return &types.QueryFeeTokensResponse{FeeTokens: feeTokens}, nil
+}
+
+func (k Keeper) DenomPoolId(ctx context.Context, req *types.QueryDenomPoolIdRequest) (*types.QueryDenomPoolIdResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	feeToken, err := k.GetFeeToken(sdkCtx, req.GetDenom())
```

Comment:
Fee token record is unique per denom? PoolID is just the pool to swap rate between alt fee token and native fee token?